portal-impl/src/com/liferay/portal/service/impl/PhoneServiceImpl.java

```diff
@@ -74,6 +92,23 @@ public Phone getPhone(long phoneId) throws PortalException {
 		return phone;
 	}
 
+	@Override
+	public Phone getPhoneByExternalReferenceCode(
+			String externalReferenceCode, long companyId)
+		throws PortalException {
+
+		Phone phone = phoneLocalService.getPhoneByExternalReferenceCode(
+			externalReferenceCode, companyId);
+
+		if (phone != null) {
```

Comment:
cannot be null (it throws an exception if doesn't exist)

Reply:
true!